src/audio/copier/copier_gain.c

```diff
@@ -15,18 +15,17 @@
 
 LOG_MODULE_DECLARE(copier, CONFIG_SOF_LOG_LEVEL);
 
-int copier_gain_set_params(struct comp_dev *dev, struct dai_data *dd)
+int copier_gain_set_params(struct comp_dev *dev, struct copier_gain_params *gain_params, uint32_t fade_period)
 {
 	struct processing_module *mod = comp_mod(dev);
 	struct copier_data *cd = module_get_private_data(mod);
 	struct ipc4_base_module_cfg *ipc4_cfg = &cd->config.base;
 	uint32_t sampling_freq = ipc4_cfg->audio_fmt.sampling_frequency;
 	uint32_t frames = sampling_freq / dev->pipeline->period;
-	uint32_t fade_period = GAIN_DEFAULT_FADE_PERIOD;
 	int ret;
 
 	/* Set basic gain parameters */
-	copier_gain_set_basic_params(dev, dd, ipc4_cfg);
+	copier_gain_set_basic_params(dev, gain_params, ipc4_cfg);
 
 	switch (dd->dai->type) {
```

Comment:
Right, so at this patch this is not going to compile. `dd` is not defined.

Reply:
for dd it looks like part of this change is placed in the next commit, I will move it to the first one

Reply:
looks that way, I'll wait for the update with the fix.